server/app/files.go

```diff
@@ -203,13 +203,33 @@ func (a *App) CopyAndUpdateCardFiles(boardID, userID string, blocks []*model.Blo
 	for _, block := range blocks {
 		if block.Type == model.TypeImage || block.Type == model.TypeAttachment {
 			if fileID, ok := block.Fields["fileId"].(string); ok {
-				blockIDs = append(blockIDs, block.ID)
-				blockPatches = append(blockPatches, model.BlockPatch{
-					UpdatedFields: map[string]interface{}{
-						"fileId": newFileNames[fileID],
-					},
-					DeletedFields: []string{"attachmentId"},
-				})
+				if err = model.ValidateFileId(fileID); err != nil {
+					blockIDs = append(blockIDs, block.ID)
+					blockPatches = append(blockPatches, model.BlockPatch{
+						UpdatedFields: map[string]interface{}{
+							"fileId": newFileNames[fileID],
+						},
+						DeletedFields: []string{"attachmentId"},
+					})
+				} else {
+					errMessage := fmt.Sprintf("invalid characters in block with key: %s, %s", block.Fields["fileId"], err)
+					return model.NewErrBadRequest(errMessage)
+				}
+			}
+
+			if attachmentID, ok := block.Fields["attachmentId"].(string); ok {
+				if err = model.ValidateFileId(attachmentID); err != nil {
+					blockIDs = append(blockIDs, block.ID)
+					blockPatches = append(blockPatches, model.BlockPatch{
+						UpdatedFields: map[string]interface{}{
+							"attachmentId": newFileNames[attachmentID],
+						},
+						DeletedFields: []string{"fileId"},
+					})
+				} else {
+					errMessage := fmt.Sprintf("invalid characters in block with key: %s, %s", block.Fields["attachmentId"], err)
```

Comment:
Same comment on generic errors

```suggestion
					errMessage := fmt.Sprintf("invalid characters in block with key: %s, %s", block.Fields[model.AttachmentId], err)
```